bundles/org.eclipse.swt.svg/src/org/eclipse/swt/svg/JSVGRasterizer.java

```diff
@@ -0,0 +1,173 @@
+			PaletteData palette = new PaletteData(colorModel.getRedMask(), colorModel.getGreenMask(),
+					colorModel.getBlueMask());
+			ImageData data = new ImageData(bufferedImage.getWidth(), bufferedImage.getHeight(),
+					colorModel.getPixelSize(), palette);
+			for (int y = 0; y < data.height; y++) {
+				for (int x = 0; x < data.width; x++) {
+					int rgb = bufferedImage.getRGB(x, y);
+					int pixel = palette.getPixel(new RGB((rgb >> 16) & 0xFF, (rgb >> 8) & 0xFF, rgb & 0xFF));
+					data.setPixel(x, y, pixel);
+					if (colorModel.hasAlpha()) {
+						data.setAlpha(x, y, (rgb >> 24) & 0xFF);
+					}
+				}
+			}
+			return data;
+		} else if (bufferedImage.getColorModel() instanceof IndexColorModel) {
+			IndexColorModel colorModel = (IndexColorModel) bufferedImage.getColorModel();
+			int size = colorModel.getMapSize();
+			byte[] reds = new byte[size];
+			byte[] greens = new byte[size];
+			byte[] blues = new byte[size];
+			colorModel.getReds(reds);
+			colorModel.getGreens(greens);
+			colorModel.getBlues(blues);
+			RGB[] rgbs = new RGB[size];
+			for (int i = 0; i < rgbs.length; i++) {
+				rgbs[i] = new RGB(reds[i] & 0xFF, greens[i] & 0xFF, blues[i] & 0xFF);
+			}
+			PaletteData palette = new PaletteData(rgbs);
+			ImageData data = new ImageData(bufferedImage.getWidth(), bufferedImage.getHeight(),
+					colorModel.getPixelSize(), palette);
+			data.transparentPixel = colorModel.getTransparentPixel();
+			WritableRaster raster = bufferedImage.getRaster();
+			int[] pixelArray = new int[1];
+			for (int y = 0; y < data.height; y++) {
+				for (int x = 0; x < data.width; x++) {
+					raster.getPixel(x, y, pixelArray);
+					data.setPixel(x, y, pixelArray[0]);
+				}
+			}
+			return data;
+		} else if (bufferedImage.getColorModel() instanceof ComponentColorModel) {
+			ComponentColorModel colorModel = (ComponentColorModel) bufferedImage.getColorModel();
+			// ASSUMES: 3 BYTE BGR IMAGE TYPE
+			PaletteData palette = new PaletteData(0x0000FF, 0x00FF00, 0xFF0000);
+			ImageData data = new ImageData(bufferedImage.getWidth(), bufferedImage.getHeight(),
+					colorModel.getPixelSize(), palette);
+			// This is valid because we are using a 3-byte Data model with no transparent
+			// pixels
+			data.transparentPixel = -1;
+			WritableRaster raster = bufferedImage.getRaster();
+			int[] pixelArray = new int[3];
+			for (int y = 0; y < data.height; y++) {
+				for (int x = 0; x < data.width; x++) {
+					raster.getPixel(x, y, pixelArray);
+					int pixel = palette.getPixel(new RGB(pixelArray[0], pixelArray[1], pixelArray[2]));
+					data.setPixel(x, y, pixel);
+				}
+			}
+			return data;
+		}
+		return null;
+	}
+
+	private boolean isSVGFile(InputStream stream) throws IOException {
+		if (stream == null) {
+			throw new IllegalArgumentException("InputStream cannot be null");
+		}
+		stream.mark(Integer.MAX_VALUE);
+		try {
+			int firstByte = stream.read();
+			return firstByte == '<';
+		} finally {
+			stream.reset();
+		}
+	}
```

Comment:
If this would be moved into the `FileFormat` we can do all special stream handling and going back and forth in the stream (if necessary) there. And this class can then just read from the stream assuming it's content is an SVG and doesn't has to care about marking, resetting etc.

And having a check if the content of an InputStream is an SVG file in a FileFormat class seems to be suitable too. And it avoids to re-implement or move this method in case there are alternative implementations at any point in time.